README.md

```diff
@@ -21,38 +21,31 @@ _Note: master represents the next major version of umzug - v3 - which is current
 
 The following example uses a Sqlite database through sequelize and persists the migration data in the database itself through the sequelize storage.
 
-* **`index.js`**:
-
 ```js
+// index.js
 const { Sequelize } = require('sequelize');
-const { Umzug, SequelizeStorage } = require('umzug');
+const { getUmzug, SequelizeStorage } = require('umzug');
 
 const sequelize = new Sequelize({ dialect: 'sqlite', storage: './db.sqlite' });
 
-const umzug = new Umzug({
-  migrations: {
-    path: './migrations',
-    params: [
-      sequelize.getQueryInterface()
-    ]
-  },
-  storage: new SequelizeStorage({ sequelize })
+const umzug = getUmzug({
+  migrations: { glob: 'migrations/*.js' },
+  context: sequelize.getQueryInterface(),
```

Comment:
I would expect `context` to be a plain object containing assorted things, like

```suggestion
  context: {
    queryInterface: sequelize.getQueryInterface(),
  }
```

what do you think?

This would allow:

```ts
const context = {
  queryInterface: sequelize.getQueryInterface()
};

export type Context = typeof context;

const umzug = getUmzug({
  migrations: { glob: 'migrations/*.js' },
  storage: new SequelizeStorage({ sequelize }),
  context
};
```

And in the migration file:

```ts
// migrations/00_initial.ts
import type { Migration } from 'umzug';
import type { Context } from '..';
// types will now be available for `queryInterface`
export const up: Migration<Context> = ({ context: queryInterface }) => queryInterface.createTable(...)
```

(of course we would have to tweak the `Migration` type to receive this generic type.

Reply:
Yeah `context` object makes sense, we can add additional options in future as well

Reply:
About the context being a plain object - that is already possible. Anything that `context` is set to will be passed as-is to the `context` property of each migration function.

Reply:
Re `export type Context` and making `Migration` generic. I would like to do something like that to. Right now it's a bit tricky because we still have the old `Migration` class, so naming gets a bit confusing. What I think we should do long term is something like

```typescript
import { getUmzug, MigrationOf } from 'umzug'

const umzug = getUmzug({
  migrations: { glob: 'migrations/*.js' },
  storage: new SequelizeStorage({ sequelize }),
  context: sequelize.getQueryInterface(),
})

export type Migration = MigrationOf<typeof umzug>
```

Then in the migration file:

```typescript
import { Migration } from '..'

export const up: Migration = ({ context }) => ...
```

That way the migrations themselves become simpler - they only need to import one thing, they don't need to use generics, and they don't have the possibility of importing `Context` from the wrong place or using `any` instead. Since there are in general far more migration files than umzug-entrypoint files, it feels more correct to have the generics live in the entrypoint.

👆 all that may need to go into a follow-up though. I've deliberately left the `Umzug` class alone as much possible. My plan was to consider the old class/constructor overloads as deprecated, then try out the new API with the sequelize CLI and the slonik migrator I maintain. If it goes well then we can move the logic "up" into the main Umzug class. Until then the `MigrationOf` in the above snippet isn't possible, since the `Umzug` class has no knowledge of `context`.